tests/KeyFrame.cpp

```diff
@@ -506,3 +513,195 @@ TEST_CASE( "Point_Vector_Constructor", "[libopenshot][keyframe]" )
+
+	CHECK(val.cx == 14.0);
+	CHECK(val.cy == 14.0);
+	CHECK(val.width == 100.0);
+	CHECK(val.height == 100.0);
+
+	val = kfb.GetBox(15);
+	
+	CHECK(val.cx == 24.0);
+	CHECK(val.cy == 24.0);
+	CHECK(val.width == 100.0);
+	CHECK(val.height == 100.0);
+
+	val = kfb.GetBox(25);
+	
+	CHECK(val.cx == 34.0);
+	CHECK(val.cy == 34.0);
+	CHECK(val.width == 100.0);
+	CHECK(val.height == 100.0);
+
+}
+
+
+TEST_CASE( "TrackedObjectBBox SetJson", "[libopenshot][keyframe]" )
+{
+	TrackedObjectBBox kfb;
+
+	kfb.AddBox(1, 10.0, 10.0, 100.0, 100.0, 0.0);
+	kfb.AddBox(10, 20.0, 20.0, 100.0, 100.0, 0.0);
+	kfb.AddBox(20, 30.0, 30.0, 100.0, 100.0, 0.0);	
+	kfb.AddBox(30, 40.0, 40.0, 100.0, 100.0, 0.0);
+
+	kfb.scale_x.AddPoint(1, 2.0);
+	kfb.scale_x.AddPoint(10, 3.0);
+
+	kfb.SetBaseFPS(Fraction(24.0, 1.0));
+
+	auto dataJSON = kfb.Json();
+	TrackedObjectBBox fromJSON_kfb;
+	fromJSON_kfb.SetJson(dataJSON);
+
+	CHECK(kfb.GetBaseFPS().num == fromJSON_kfb.GetBaseFPS().num);
+	
+	double time_kfb = kfb.FrameNToTime(1, 1.0);
+	double time_fromJSON_kfb = fromJSON_kfb.FrameNToTime(1, 1.0);
+	CHECK(time_kfb == time_fromJSON_kfb);
+
+	BBox kfb_bbox =  kfb.BoxVec[time_kfb];
+	BBox fromJSON_bbox = fromJSON_kfb.BoxVec[time_fromJSON_kfb];
+
+	CHECK(kfb_bbox.cx == fromJSON_bbox.cx);
+	CHECK(kfb_bbox.cy == fromJSON_bbox.cy);
+	CHECK(kfb_bbox.width == fromJSON_bbox.width);
+	CHECK(kfb_bbox.height == fromJSON_bbox.height);
+	CHECK(kfb_bbox.angle == fromJSON_bbox.angle);
+}
+
+TEST_CASE( "TrackedObjectBBox scaling", "[libopenshot][keyframe]" )
+{
+	TrackedObjectBBox kfb;
+
+	kfb.AddBox(1, 10.0, 10.0, 10.0, 10.0, 0.0);
+	kfb.scale_x.AddPoint(1.0, 2.0);
+	kfb.scale_y.AddPoint(1.0, 3.0);
+	
+	BBox bbox = kfb.GetBox(1);
+	
+	CHECK(bbox.width == 20.0);
+	CHECK(bbox.height == 30.0);
+}
+
+TEST_CASE( "AttachToObject", "[libopenshot][keyframe]" )
+{
+	std::stringstream path1, path2;
+	path1 << TEST_MEDIA_PATH << "test.avi";
+	path2 << TEST_MEDIA_PATH << "run.mp4";
+
+	// Create Timelime
+	Timeline t(1280, 720, Fraction(25,1), 44100, 2, ChannelLayout::LAYOUT_STEREO);
+
+	// Create Clip and add it to the Timeline
+	Clip clip(new FFmpegReader(path1.str()));
+	clip.Id("AAAA1234");
+
+	// Create a child clip and add it to the Timeline
+	Clip childClip(new FFmpegReader(path2.str()));
+	childClip.Id("CHILD123");
+
+	// Add clips to timeline
+	t.AddClip(&childClip);
+	t.AddClip(&clip);
+
+	// Create tracker and add it to clip
+	Tracker tracker;
+	clip.AddEffect(&tracker);
+
+	// Save a pointer to trackedData
+	std::shared_ptr<TrackedObjectBBox> trackedData = tracker.trackedData;
+
+	// Change trackedData scale
+	trackedData->scale_x.AddPoint(1, 2.0);
+	CHECK(trackedData->scale_x.GetValue(1) == 2.0);
+
+	// Tracked Data JSON
+	auto trackedDataJson = trackedData->JsonValue();
+
+	// Get and cast the trakcedObject
+	auto trackedObject_base = t.GetTrackedObject("None");
+	std::shared_ptr<TrackedObjectBBox> trackedObject = std::static_pointer_cast<TrackedObjectBBox>(trackedObject_base);
```

Comment:
That's a really long line, which I feel would be cleaner split in two.

Also, [according to cppreference](https://en.cppreference.com/w/cpp/memory/shared_ptr/pointer_cast#Example) `std::static_pointer_cast` is for casting _up_ the inheritance tree, while `std::dynamic_pointer_cast` should be used when casting _down_ or _sideways_. So, to cast a `TrackedObjectBase` to a `TrackedObjectBBox`, this should be:
```suggestion
	auto trackedObject = std::make_shared<TrackedObjectBBox>();
	trackedObject = std::dynamic_pointer_cast<TrackedObjectBBox>(trackedObject_base);
```